daml/Marketplace/Issuance/Service.daml

```diff
@@ -85,27 +84,22 @@ template Service
         with
           createIssuanceRequestCid : ContractId CreateIssuanceRequest
         do
-          CreateIssuanceRequest{issuanceId; assetId; accountId; quantity} <- fetch createIssuanceRequestCid
-          archive createIssuanceRequestCid
-
-          Some _ <- lookupByKey @AssetDescription assetId
+          CreateIssuanceRequest{issuanceId; assetId; account; quantity} <- fetchAndArchive createIssuanceRequestCid
           issuanceCid <- create Issuance with ..
 
+          (_, AssetDescription{registrar}) <- fetchByKey @AssetDescription assetId
           let asset = Asset with id = assetId; quantity
-          depositCid <- exerciseByKey @AssetSettlementRule accountId AssetSettlement_Credit with ctrl = provider; ..
-          deposit    <- fetch depositCid
-          depositCid <- exercise depositCid AssetDeposit.AssetDeposit_SetObservers with newObservers = deposit.asset.id.signatories `union` deposit.observers
+          depositCid <- create AssetDeposit with account; asset; lockers = empty; observers = fromList [ registrar ]
+
```

Comment:
Same comment as in `CustodyService` - should the observer be the `issuer` here?

Reply:
Updated in the latest commit - Yes, this is so the `issuer` has visibility on parties which holds assets the `issuer` has created.